### PR TITLE
Anomaly CI: Add action for task-sdk, add code coverage to workflow, add tests

### LINKED ISSUE #52: Research Best Coverage Metric
### Body

A summary of the current state of testing in the pre-existing system you are enhancing; for automatable tests, **report a coverage metric**

### Committer

- [x] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### LINKED ISSUE #53: Add Basic Anomaly SDK Testing
### Body

Build out test coverage for our changes to the anomaly sdk, leveraging the CI merge and existing airflow test framework

### Committer

- [x] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by Ethan-Russell**
# Summary
* Added to anomaly ci: `.github/workflows/ci-anomaly.yml`
  * Added an action for task-sdk tests.
  * Added code coverage reporting to task-sdk, backend, and frontend tests
* Made a test script for backend: `airflow-core/tests/unit/api_fastapi/core_api/routes/public/test_task_instance_anomaly.py`
* Made a test script for task-sdk: `task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py`
* Made a test script for backend: `airflow-core/tests/unit/api_fastapi/core_api/routes/public/test_task_instance_anomaly.py`

Closes:
* #53

Works towards:
* #52 

---

##### Was generative AI tooling used to co-author this PR?

- [ ] Yes (please specify the tool below)

---

* Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information. Note: commit author/co-author name and email in commits become permanently public when merged.
* For fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
* When adding dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
* For significant user-facing changes create newsfragment: `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

### CODE REVIEW

File: airflow-core/src/airflow/ui/src/pages/AnomalyDashboard/AnomalyDashboard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+describe("AnomalyDashboard", () => {
+  it("renders header, stat cards, and empty recent-anomalies table", () => {
+    render(<AnomalyDashboard />, { wrapper: Wrapper });
+
+    expect(screen.getByText("Anomalies")).toBeInTheDocument();
+    expect(screen.getByText("Anomaly dashboard description")).toBeInTheDocument();
+    expect(screen.getByText("Last 24h")).toBeInTheDocument();
+    expect(screen.getByText("Tasks monitored")).toBeInTheDocument();
+    expect(screen.getByText("Algorithm")).toBeInTheDocument();
```
Comment (yixuanyang123) [inline airflow-core/src/airflow/ui/src/pages/AnomalyDashboard/AnomalyDashboard.test.tsx:89]: Just wanted to flag something I noticed: the test assertions are looking for strings like "Anomalies", "Last 24h", "Algorithm", and "No anomalies found", but the current AnomalyDashboard.tsx has hardcoded strings like "Task Performance Anomalies", "Anomalies (last 24h)", "Detection algorithm", etc.

Am I missing something, or is this test written against a planned i18n version of the component that hasn't landed yet? If so, I'd expect this to fail right now.

If the intent is to add i18n support later, it might be worth either:
1. Temporarily matching the test assertions to the current hardcoded strings, or
2. Noting in the PR that this test is expected to be broken until the i18n work is done.

Happy to be corrected if I'm misreading the setup!

File: task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    def test_anomaly_when_latest_below_min(self) -> None:
+        algo = ThresholdAnomaly(min_runtime=5.0, max_runtime=100.0)
+        result = algo([10.0, 3.0])
+        assert result.is_anomaly is True
+
+    def test_anomaly_when_latest_above_max(self) -> None:
+        algo = ThresholdAnomaly(min_runtime=0.0, max_runtime=1.0)
+        result = algo([0.5, 2.0])
+        assert result.is_anomaly is True
```
Comment (yixuanyang123) [inline task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py:43]: The tests here cover AlwaysAnomaly and ThresholdAnomaly nicely! Was there a reason the AnomalyDetector class wasn't included? It has quite a bit of logic (the min_runs skip, max_runs truncation, the supervisor comms path) that seems like it could be tricky to get right.

File: task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class TestAlwaysAnomaly:
+    def test_flags_final_runtime(self) -> None:
+        result = AlwaysAnomaly()([1.0, 2.0])
+        assert isinstance(result, AnomalyResult)
+        assert result.is_anomaly is True
+
+
+class TestThresholdAnomaly:
```
Comment (yixuanyang123) [inline task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py:29]: For the ThresholdAnomaly test, might also be worth adding a test for when the value lands exactly on the boundary (equal to min or max), just to confirm the >= / <= behavior is intentional.

File: task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py (hunk 2 above)
Comment (Ethan-Russell) [reply task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py:43]: Good catch! This is just an initial test file, with the intention of making it more robust and including the AnomalyDetector tests down the road.

File: task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py (hunk 3 above)
Comment (Ethan-Russell) [reply task-sdk/tests/task_sdk/definitions/test_anomaly_detection.py:29]: This would be a great thing to test.  I'll spin up a new issue for additional tests to add in the task-sdk: #65

File: airflow-core/src/airflow/ui/src/pages/AnomalyDashboard/AnomalyDashboard.test.tsx (hunk 1 above)
Comment (Ethan-Russell) [reply airflow-core/src/airflow/ui/src/pages/AnomalyDashboard/AnomalyDashboard.test.tsx:89]: Thanks for raising that. This test uses a local i18n instance and simply checks to make sure that the strings in that instance are making it to the dashboard correctly.  The tests in this PR are meant to be preliminary, and should be replaced or reinforced as we advance our testing.

**Comment by yixuanyang123** [comment]
Looks good to me. Just merged to dev
